frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -169,10 +170,12 @@ export default ({
     }
   },
   created () {
-    // TODO #492 create a global Vue Responsive just for media queries.
-    const mediaIsPhone = window.matchMedia('screen and (max-width: 639px)')
-    this.config.isPhone = mediaIsPhone.matches
-    mediaIsPhone.onchange = (e) => { this.config.isPhone = e.matches }
+    // TODO: #492 create a global Vue Responsive just for media queries.
+    this.config.mediaIsPhone = window.matchMedia('screen and (max-width: 639px)')
+    this.config.mediaIsPhone.onchange = (e) => {
+      this.config.isPhone = e.matches
+    }
+    this.config.isPhone = this.config.mediaIsPhone.matches
```

Comment:
Why is `mediaIsPhone` being added to the config? Now there's two config variables to keep track of when we just want to know if we're a mobile device, right? So wouldn't it be better to just have one variable?

Reply:
@taoeffect 
@taoeffect The purpose of this change is to destroy the `onchange` matchMedia handler along with destruction of `ChatMain.vue` (L188 below). So I wanted to expose this matchMedia instance as a component level variable instead of a local variable in `created()` hook. But now that you've mentioned that, it looks like it's not necessary to define it in `config` object. I could just attach it as `this.matchMediaPhone`. Does it sound good?

Reply:
Ah I see, yes that sounds good. Nice catch! And I like `matchMediaPhone` as a variable name better too 👍

Reply:
Should we declare the `matchMediaPhone` variable inside the `data`?

Reply:
@Silver-IT I think it's not necessary in this case because, (unlike `this.config.isPhone`) it's not the kind of state that is linked to the DOM update, let alone the fact that it's only used in two life cycle hooks in the component. Thanks.